app/lib/actions/pulp/repository/refresh.rb

```diff
@@ -39,7 +39,8 @@ def update_or_associate_importer(capsule_id, repository, repository_details)
         def update_or_associate_distributors(capsule_id, repository, repository_details)
           concurrence do
             existing_distributors = repository_details["distributors"]
-            repository.generate_distributors(capsule_id.present?).each do |distributor|
+            capsule = capsule_id ? ::Katello::CapsuleContent.new(SmartProxy.find(capsule_id)) : nil
```

Comment:
Maybe I'm just misunderstanding, but why is this a  CapsuleContent object? Seems like you're just calling capsule_content.capsule in the generate_distributor method, which is just exactly the SmartProxy object you already have here